torch/_numpy/_funcs_impl.py

```diff
@@ -359,22 +359,25 @@ def arange(
 
     # the dtype of the result
     if dtype is None:
-        dtype = _dtypes_impl.default_dtypes().int_dtype
-    # XXX: default values do not get normalized
-    start, stop, step = (_util._coerce_to_tensor(x) for x in (start, stop, step))
-
-    dummy = torch.empty(1, dtype=dtype)
-    target_dtype = _dtypes_impl.result_type_impl(start, stop, step, dummy)
+        dtype = (
+            _dtypes_impl.default_dtypes().float_dtype
+            if any(_dtypes_impl.is_float_or_fp_tensor(x) for x in (start, stop, step))
+            else _dtypes_impl.default_dtypes().int_dtype
+        )
+    work_dtype = torch.float64 if dtype.is_complex else dtype
 
-    # work around RuntimeError: "arange_cpu" not implemented for 'ComplexFloat'
-    work_dtype = torch.float64 if target_dtype.is_complex else target_dtype
+    # work around RuntimeError: "lt_cpu" not implemented for 'ComplexFloat'
+    if any(_dtypes_impl.is_complex_or_complex_tensor(x) for x in (start, stop, step)):
+        work_dtype = torch.float64
+        dtype = torch.complex128
+        start, stop, step = float(start), float(stop), float(step)
```

Comment:
This does not work?
```python
>>> float(1+1j)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: float() argument must be a string or a real number, not 'complex'
```

Let's remove this case and let's just handle the `dtype=complex` path, that should be enough to fix the tests

Reply:
Let's raise NotImplementedError --- can't just remove, because

```
In [1]: import numpy as np

In [2]: np.arange(1+1j)
Out[2]: array([0.+0.j])

In [3]: import torch._numpy as tnp

In [4]: tnp.arange(5+2j)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[4], line 1
----> 1 tnp.arange(5+2j)

File ~/repos/pytorch/torch/_numpy/_normalizations.py:222, in normalizer.<locals>.normalizer_inner.<locals>.wrapped(*args, **kwds)
    209     args = (
    210         tuple(
    211             maybe_normalize(arg, parm)
   (...)
    214         + args[len(params.values()) :]
    215     )
    217 kwds = {
    218     name: maybe_normalize(arg, params[name]) if name in params else arg
    219     for name, arg in kwds.items()
    220 }
--> 222 result = func(*args, **kwds)
    224 # keepdims
    225 bound_args = None

File ~/repos/pytorch/torch/_numpy/_funcs_impl.py:377, in arange(start, stop, step, dtype, like)
    367 work_dtype = torch.float64 if dtype.is_complex else dtype
    369 '''
    370 # work around RuntimeError: "lt_cpu" not implemented for 'ComplexFloat'
    371 if any(_dtypes_impl.is_complex_or_complex_tensor(x) for x in (start, stop, step)):
   (...)
    374     start, stop, step = float(start), float(stop), float(step)
    375 '''
--> 377 if (step > 0 and start > stop) or (step < 0 and start < stop):
    378     # empty range
    379     return torch.empty(0, dtype=dtype)
    381 result = torch.arange(start, stop, step, dtype=work_dtype)

TypeError: '>' not supported between instances of 'int' and 'complex'
```

Reply:
Even better, yep